graphstorm-processing/tests/test_dist_label_loader.py

```diff
@@ -0,0 +1,127 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from typing import List
+
+import numpy as np
+
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructField, StructType, StringType
+
+
+from graphstorm_processing.data_transformations.dist_label_loader import DistLabelLoader
+from graphstorm_processing.config.label_config_base import LabelConfig
+
+
+def test_dist_classification_label(spark: SparkSession, input_df: DataFrame, check_df_schema):
+    """Test classification label generation"""
+    label_col = "name"
+    classification_config = {
+        "column": "name",
+        "type": "classification",
+        "split_rate": {"train": 0.8, "val": 0.2, "test": 0.0},
+    }
+
+    label_transformer = DistLabelLoader(LabelConfig(classification_config), spark)
+
+    transformed_labels = label_transformer.process_label(input_df)
+
+    label_map = label_transformer.label_map
+
+    assert set(label_map.keys()) == {"mark", "john", "tara", "kate", "jen"}
+
+    check_df_schema(transformed_labels)
+
+    transformed_rows = transformed_labels.collect()
+
+    row_val_set = set()
+    for row in transformed_rows:
+        row_val_set.add(row[label_col])
+
+    # We expect 5 distinct int label values as each element is unique
+    assert len(row_val_set) == 5
+    for val in row_val_set:
+        assert isinstance(val, int)
+
+
+def test_dist_regression_label(spark: SparkSession, input_df: DataFrame, check_df_schema):
+    """Test regression label generation"""
+    label_col = "age"
+    regression_config = {
+        "column": label_col,
+        "type": "regression",
+        "split_rate": {"train": 1.0, "val": 0.0, "test": 0.0},
+    }
+
+    label_transformer = DistLabelLoader(LabelConfig(regression_config), spark)
+
+    # Regression labels are expected to be float values
+    input_df = input_df.withColumn(label_col, input_df[label_col].cast("float"))
+
+    transformed_labels = label_transformer.process_label(input_df)
+
+    check_df_schema(transformed_labels)
+
+    transformed_rows = transformed_labels.collect()
+
+    row_val_set = set()
+    for row in transformed_rows:
+        row_val_set.add(row[label_col])
+
+    # We expect 4 distinct label values
+    assert len(row_val_set) == 4
+    for val in row_val_set:
+        assert isinstance(val, (float, type(None)))
```

Comment:
Can we also check the mask?

Reply:
Mask creation is tested in https://github.com/awslabs/graphstorm/blob/main/graphstorm-processing/tests/test_dist_heterogenous_loader.py#L443-L525. We could also test it here, but then we would breaking the single-purpose of the unit test.

Reply:
If it is already tested. I think it is fine.